src/index.js

```diff
@@ -0,0 +1,159 @@
+export default class toColor {
+  HUE_MAX = 360;
+
+  constructor(seed, options) {
+    this.options = options || {};
+    this.seed = typeof seed === 'string' ? this._stringToInteger(seed) : seed;
+    this.known = [];
+  }
+
+  getColor() {
+    const { distance, knownMax } = this.options;
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const PASSABLE_DISTANCE = typeof distance !== 'undefined' ? distance : 37; 
+    const KNOWN_MAX = typeof knownMax !== 'undefined' ? knownMax : 20; 
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+      this.known.length <= KNOWN_MAX &&
+      this.known.some(v => this._euclideanDistance(v, hsl) < PASSABLE_DISTANCE)) {
+      return this.getColor();
+    } else {
+      this.known.push([...hsl]);
+      return this._colorWithModifiers(h, s, b) 
+    }
+  }
+
+  _colorWithModifiers = (h, s, b) => {
```

Comment:
It makes sense that we modify after rather than before getting color to ensure same colors are returned for same seed regardless of modifiers but I think you should note this with a comment.